blockchain/source/test/profileRegistry.js

```diff
@@ -534,4 +536,104 @@ contract('ProfileRegistry', async function (accounts) {
             });
         });
     });
+
+    describe('AddSonmValidator', function () {
+        before(async function () {
+            registry = await ProfileRegistry.new();
+        });
+
+        it('should set new validator with level -1', async function () {
+            await registry.AddSonmValidator(specialValidator, { from: owner });
+            let lvl = (await registry.GetValidatorLevel(specialValidator)).toNumber();
+            assert.equal(lvl, -1);
+        });
+
+        after(async function () {
+            await registry.RemoveSonmValidator(specialValidator, { from: owner });
+        });
+    });
+
+    describe('RemoveSonmValidator', function () {
+        before(async function () {
+            registry = await ProfileRegistry.new();
+
+            await registry.AddSonmValidator(specialValidator, { from: owner });
+        });
+
+        it('should set remove validator with level -1', async function () {
+            await registry.RemoveSonmValidator(specialValidator, { from: owner });
+            let lvl = (await registry.GetValidatorLevel(specialValidator)).toNumber();
+            assert.equal(lvl, 0);
+        });
+    });
+
+    describe('when contract is paused', function () {
+        describe('Pause calling should spend `Paused` event', function () {
+            let tx;
+
+            before(async function () {
+                registry = await ProfileRegistry.new();
+            });
+
+            it('should pause market', async function () {
+                tx = await registry.pause();
+            });
+
+            it('should spend `Pause` event', function () {
+                assert.equal(tx.logs.length, 1);
+                assert.equal(tx.logs[0].event, 'Pause');
+            });
+        });
+
+        describe('AddValidator', function () {
+            before(async function () {
+                registry = await ProfileRegistry.new();
+                await registry.pause();
+            });
+
+            it('should revert', async function () {
+                await assertRevert(registry.AddValidator(testValidator, testLevel, { from: owner }));
+            });
+        });
+
+        describe('RemoveValidator', function () {
+            before(async function () {
+                registry = await ProfileRegistry.new();
+                await registry.AddValidator(testValidator, testLevel, { from: owner });
+                await registry.pause();
+            });
+
+            it('remove validator and spend `ValidatorDeleted` event', async function () {
```

Comment:
but it not checks for validator removal and event emitting.

Reply:
thx, fixed